### DISCUSSION TITLE
SPI Analyzer

### DISCUSSION THREAD

**Discussion by psylapse** [gusmanb/logicanalyzer#71] [General]
Really enjoying the analyser, for which many thanks.  I am having difficulty launching the SPI analyzer, both under Windows and Linux, so having to export CSV to pulseview.  The DLL launches and asks the right questions (Clk, phase etc) but after accept nothing is shown. Are there known limits (eg sample size) or am I doing something wrong.  I note that there are screen snaps of the software doing a SPI decode and the other two (I2C and serial) work just fine.  Many thanks

**Comment by gusmanb** [comment]
There are no limits and the usage should be straight-forward. If you save a capture and send me it with the params you used for the analysis I can check if there is any problem.

Cheers.

**Comment by psylapse** [comment]
Many thanks for your help  .... and so rapidly.
The screen snap below shows most of the parameters (off screen is CS none and Clk channel 2) . Clicking on accept just drops me back to the screen below, with no annotation added (as is the case with eg I2C)
<img width="2880" alt="LASPI" src="https://github.com/gusmanb/logicanalyzer/assets/61235234/88c36823-1b73-490d-9dbc-a5c4452ad875">

keep up the good work

**Reply by gusmanb** [reply]
Sorry, I think there is a misunderstanding, I need the captured data to debug the analyzer, not a capture of the screen :)
You can save the capture from the File menu: File->Save, that will save the data with all the capture parameters.

**Reply by psylapse** [reply]
My apologies,  I cannot attach the lac file directly, github will not let me.

...hope the zip is bit perfect !

[cc2500 start.zip](https://github.com/gusmanb/logicanalyzer/files/12222979/cc2500.start.zip)

**Reply by gusmanb** [reply]
No worries :)

Ok, found the problem, it's related to not setting the CS channel, the analyzer should support it but because an update to the settings window it was messed and it thought that were no select data. I have packaged a version with the problem corrected, you can download it here: https://mega.nz/file/aPJhDKjQ#YBxTszd2MfXafFuDwj-gwBRrW2MwBV_03awZHHw5sBc

I must warn you two things. 

First, the version that I have uploaded is the new V5.0, you will need to update the firmware of the analyzer to V5.0 also, in this version the burst mode has been introduced and I had to change the protocol to include the burst data, if you try to connect an analyzer with an older version it will refuse to connect.

Second, in your capture there is an spurious pulse at the beginning of the capture on all channels, so as you are not providing the CS channel it is interpreted as a clock pulse and the data will not be correct. If you have a CS line I would encourage you to capture it so the analyzer will know when the device is active or not. If you cannot capture it then you will need to delete the samples with that pulse in order to correct the data. Also, if you really don't have a CS line and your slave device is always selected and you're having problems, that may be the root cause of it ;)

I will include the patch on the next release.

Cheers!

**Reply by psylapse** [reply]
Fantastic.  I have downloaded the version 5 firmware as well and will setup another pico with it (when I can find where I put it).  You are of course correct, much better to have CS as well, but I had enough difficulty soldering on three wires onto 3 legs , all within 1mm!  When I was young (long long ago)  ICs came exclusively with legs on a 2.54mm pitch.  So this project will definitely make it into my long term tool box.  For which many thanks.

And good luck with a long list of other projects I see against your name
